File: app/src/main/java/com/sjl/bookmark/ui/presenter/ExpressCompanyPresenter.java
package com.sjl.bookmark.ui.presenter;

import com.google.gson.Gson;
import com.google.gson.JsonArray;
import com.google.gson.JsonElement;
import com.google.gson.JsonParser;
import com.sjl.bookmark.entity.ExpressCompany;
import com.sjl.bookmark.ui.contract.ExpressCompanyContract;
import com.sjl.core.util.log.LogUtils;

import java.io.IOException;
import java.io.InputStream;
import java.util.ArrayList;
import java.util.List;

/**
 * TODO
 *
 * @author Kelly
 * @version 1.0.0
 * @filename ExpressCompanyPresenter.java
 * @time 2018/4/29 20:37
 * @copyright(C) 2018 song
 */
public class ExpressCompanyPresenter extends ExpressCompanyContract.Presenter {


    @Override
    public List<ExpressCompany> initCompany() {
        List<ExpressCompany> companyList = new ArrayList<>();
        try {
            InputStream is = mContext.getAssets().open("company.json");
            int size = is.available();
            byte[] buffer = new byte[size];
            is.read(buffer);
            is.close();
            String json = new String(buffer);

            Gson gson = new Gson();
            JsonParser parser = new JsonParser();
            JsonArray jArray = parser.parse(json).getAsJsonArray();
            for (JsonElement obj : jArray) {
                ExpressCompany company = gson.fromJson(obj, ExpressCompany.class);
                companyList.add(company);
            }
        } catch (IOException e) {
            LogUtils.e("解析快递公司json文件异常",e);
        }
        return companyList;
    }
}

File: app/src/main/java/com/sjl/bookmark/ui/presenter/ExpressCompanyPresenter.kt
package com.sjl.bookmark.ui.presenter

import com.google.gson.Gson
import com.google.gson.JsonArray
import com.google.gson.JsonElement
import com.google.gson.JsonParser
import com.sjl.bookmark.entity.ExpressCompany
import com.sjl.bookmark.ui.contract.ExpressCompanyContract
import com.sjl.core.util.log.LogUtils
import java.io.IOException
import java.io.InputStream
import java.util.*

/**
 * TODO
 *
 * @author Kelly
 * @version 1.0.0
 * @filename ExpressCompanyPresenter.java
 * @time 2018/4/29 20:37
 * @copyright(C) 2018 song
 */
class ExpressCompanyPresenter : ExpressCompanyContract.Presenter() {
    override fun initCompany(): List<ExpressCompany> {
        val companyList: MutableList<ExpressCompany> = ArrayList()
        try {
            val `is`: InputStream = mContext.assets.open("company.json")
            val size: Int = `is`.available()
            val buffer: ByteArray = ByteArray(size)
            `is`.read(buffer)
            `is`.close()
            val json: String = String(buffer)
            val gson: Gson = Gson()
            val parser: JsonParser = JsonParser()
            val jArray: JsonArray = parser.parse(json).asJsonArray
            for (obj: JsonElement? in jArray) {
                val company: ExpressCompany = gson.fromJson(obj, ExpressCompany::class.java)
                companyList.add(company)
            }
        } catch (e: IOException) {
            LogUtils.e("解析快递公司json文件异常", e)
        }
        return companyList
    }
}